src/indicator/assuan.ts

```diff
@@ -0,0 +1,366 @@
+        const match = regex.exec(payload);
+        if (!match || !match.groups) {
+            throw new Error("fail to parse inquire response");
+        }
+        return new ResponseInquire(match.groups["keyword"], match.groups["parameters"]);
+    }
+}
+
+function splitLines(data: Buffer): Array<Buffer> {
+    const result: Buffer[] = [];
+    while (data.length > 0) {
+        const index = data.indexOf('\n', 0, 'utf8');
+        if (index === -1) {
+            throw new Error("the input data is contains no \\n character");
+        }
+        result.push(data.subarray(0, index));
+        data = data.subarray(index + 1, data.length);
+    }
+    return result;
+}
+
+function sleep(ms: number): Promise<void> {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+interface Logger {
+    log(message: string): void;
+}
+
+/**
+ * The AssuanClient class is a helper client for Assuan Protocol.
+ */
+class AssuanClient {
+    #logger: Logger;
+    #socket: net.Socket;
+
+    #responseLines: Buffer[] = [];
+
+    #socketErrorBuffer: Error[] = [];
+    #isConnected = false;
+    #isClosed = false;
+
+    /**
+     * Construct a client for Assuan Protocol
+     *
+     * @remarks User should wait initialize() to complete before sending any command.
+     *
+     * @param logger - An object which implement Console interface for debug message.
+     * @param socketPath - The file path to GnuPG unix socket.
+     */
+    constructor(logger: Logger, socketPath: string) {
+        this.#logger = logger;
+
+        this.#socket = net.createConnection(socketPath, () => {
+            this.#isConnected = true;
+        });
+
+        this.#socket.on('data', (data: Buffer) => {
+            const lines = splitLines(data);
+            for (const line of lines) {
+                this.#responseLines.push(line);
+                this.#logger.log('Recv: ' + line.toString('utf8'));
+            }
+        });
+
+        this.#socket.on('error', (error: Error) => {
+            this.#socketErrorBuffer.push(error);
+        });
+
+        this.#socket.on('close', () => {
+            this.#isClosed = true;
+        });
+    }
+
+    /**
+     * Wait fo for the underline connection to be established.
+     *
+     * @returns Nothing
+     */
+    async initialize(): Promise<void> {
+        for (; ;) {
+            if (!this.#isConnected) {
+                await sleep(0);
+            }
+            return;
+        }
+    }
+
+    /**
+     * Close the underline connection.
+     *
+     * @returns Nothing
+     */
+    async dispose(): Promise<void> {
+        this.#socket.destroy();
+
+        for (; ;) {
+            if (!this.#isClosed) {
+                await sleep(0);
+                continue;
+            }
+            return;
+        }
+    }
+
+    async sendRequest(request: Request): Promise<void> {
+        this.checkError();
+
+        const line = request.toBytes();
+        this.#logger.log('Send:' + line.toString('utf8'));
```

Comment:
In production usage `verbose` might be better
```suggestion
        this.#logger.verbose('Assuan data sent:' + line.toString('utf8'));
```